### ISSUE TITLE
Style/VariableName doesn't work for method arguments

### ISSUE THREAD

**Issue by kjvarga** [rubocop-hq/rubocop#7275]
The `Style/VariableName` rule isn't catching bad variable names (e.g. camelCase when snake_case is configured) for variables passed to methods as arguments.  

It also doesn't catch bad `let` variable definitions, thought I think that is understandable, but would be a great feature.

The problem then is that in a lot of specs we define the variable using `let` and then it gets passed to a method when it's used, but this rule isn't catching either of those scenarios, leaving a big blind spot. 

It does catch bad parameters in a method definition, and bad variable assignments.

Tested with Rubocop 0.74.0

## Expected behavior

When configured to use `snake_case`, the `Style/VariableName` rule should catch usage such as:
```
my_method(badVariableName, anotherBadVariableName)
```

## Actual behavior

No Rubocop violations are found.

```
$ rubocop test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Steps to reproduce the problem

File: test.rb
```
# frozen_string_literal: true

let(:badVariable) { true }

def print_name(user_name)
  puts user_name
end

print_name(badVariable)
```
`rubocop test.rb` (probably need RSpec for `let`)

## RuboCop version

```
$ rubocop -V
0.74.0 (using Parser 2.6.3.0, running on ruby 2.3.7 universal.x86_64-darwin18)
```

**Comment by koic** [comment]
Thanks for your feedback. I opened  a PR #7278 for this false negative of method argument names when invoking a method.

OTOH, `let` is a feature specific to RSpec. Can you feedback to RuboCop RSpec about when using `let`?
https://github.com/rubocop-hq/rubocop-rspec/issues

/cc @bquorning

**Comment by kjvarga** [comment]
Awesome!  Will do.
